### ISSUE TITLE
Blade servers merging together on import

### ISSUE THREAD

**Issue by justinhippen** [bb-Ricardo/netbox-sync#41]
When I run the sync, my 12 UCS blade servers end up merging into only 2 server objects in Netbox. Below is the DEBUG2 for one of the overwrites. 
![2021-02-12 08_22_46-uslis-p-esx011 - Copy](https://user-images.githubusercontent.com/20194814/107780329-2c961300-6d0c-11eb-887b-1a1f2fd1133e.png)

```
2021-02-12 13:56:38,840 - DEBUG2: Parsing vCenter host: uslis-p-esx001.example.com
2021-02-12 13:56:38,845 - DEBUG2: Trying to find site name for device 'uslis-p-esx001.example.com'
2021-02-12 13:56:38,845 - DEBUG2: Found a match (.*lis.*) for uslis-p-esx001.example.com, using site 'Chicago, IL'
2021-02-12 13:56:39,858 - DEBUG2: Found host vSwitch vSwitch0
2021-02-12 13:56:39,858 - DEBUG2: Found host vSwitch iScsiBootvSwitch
2021-02-12 13:56:39,858 - DEBUG2: Found host vSwitch vSwitch1
2021-02-12 13:56:40,762 - DEBUG2: Found host proxySwitch DataCluster_dvSwitch
2021-02-12 13:56:41,690 - DEBUG2: Found host portGroup Fault Tolerance VMkernel Port
2021-02-12 13:56:41,690 - DEBUG2: Found host portGroup vMotion VMkernel Port
2021-02-12 13:56:41,690 - DEBUG2: Found host portGroup Management VMKernel Port
2021-02-12 13:56:41,690 - DEBUG2: Found host portGroup VM - iSCSI-A
2021-02-12 13:56:41,690 - DEBUG2: Found host portGroup iSCSI-A
2021-02-12 13:56:41,691 - DEBUG2: Found host portGroup VM - iSCSI-B
2021-02-12 13:56:41,691 - DEBUG2: Found host portGroup iSCSI-B
2021-02-12 13:56:42,571 - DEBUG2: Parsing PhysicalNic: vmnic0
2021-02-12 13:56:42,572 - DEBUG2: Trying to find a VLAN based on the VLAN id '27'
2021-02-12 13:56:42,573 - DEBUG2: Found a exact matching VLAN object: 27 (Chicago, IL)
2021-02-12 13:56:42,573 - DEBUG2: Trying to find a VLAN based on the VLAN id '26'
2021-02-12 13:56:42,575 - DEBUG2: Found a exact matching VLAN object: 26 (Chicago, IL)
2021-02-12 13:56:42,575 - DEBUG2: Trying to find a VLAN based on the VLAN id '20'
2021-02-12 13:56:42,577 - DEBUG2: Found a exact matching VLAN object: 20 (Chicago, IL)
2021-02-12 13:56:42,577 - DEBUG2: Parsing PhysicalNic: vmnic1
2021-02-12 13:56:42,577 - DEBUG2: Trying to find a VLAN based on the VLAN id '27'
2021-02-12 13:56:42,578 - DEBUG2: Found a exact matching VLAN object: 27 (Chicago, IL)
2021-02-12 13:56:42,579 - DEBUG2: Trying to find a VLAN based on the VLAN id '26'
2021-02-12 13:56:42,580 - DEBUG2: Found a exact matching VLAN object: 26 (Chicago, IL)
2021-02-12 13:56:42,580 - DEBUG2: Trying to find a VLAN based on the VLAN id '20'
2021-02-12 13:56:42,582 - DEBUG2: Found a exact matching VLAN object: 20 (Chicago, IL)
2021-02-12 13:56:42,582 - DEBUG2: Parsing PhysicalNic: vmnic2
2021-02-12 13:56:42,582 - DEBUG2: Parsing PhysicalNic: vmnic3
2021-02-12 13:56:42,582 - DEBUG2: Parsing PhysicalNic: vmnic4
2021-02-12 13:56:42,582 - DEBUG2: Parsing PhysicalNic: vmnic5
2021-02-12 13:56:43,456 - DEBUG2: Parsing HostVirtualNic: vmk0
2021-02-12 13:56:43,457 - DEBUG2: Trying to find a VLAN based on the VLAN id '20'
2021-02-12 13:56:43,458 - DEBUG2: Found a exact matching VLAN object: 20 (Chicago, IL)
2021-02-12 13:56:43,458 - DEBUG: IP address 'fe80::225:b5ff:fe01:a17/64' for vmk0 is a link local address. Skipping.
2021-02-12 13:56:43,459 - DEBUG2: Parsing HostVirtualNic: vmk1
2021-02-12 13:56:43,459 - DEBUG: IP address 'fe80::225:b5ff:feaa:a000/64' for vmk1 is a link local address. Skipping.
2021-02-12 13:56:43,459 - DEBUG2: Parsing HostVirtualNic: vmk2
2021-02-12 13:56:43,459 - DEBUG: IP address 'fe80::250:56ff:fe6b:e6e/64' for vmk2 is a link local address. Skipping.
2021-02-12 13:56:43,459 - DEBUG2: Parsing HostVirtualNic: vmk3
2021-02-12 13:56:43,459 - DEBUG2: Trying to find a VLAN based on the VLAN id '26'
2021-02-12 13:56:43,461 - DEBUG2: Found a exact matching VLAN object: 26 (Chicago, IL)
2021-02-12 13:56:43,461 - DEBUG: IP address 'fe80::250:56ff:fe6f:5e3d/64' for vmk3 is a link local address. Skipping.
2021-02-12 13:56:43,461 - DEBUG2: Parsing HostVirtualNic: vmk4
2021-02-12 13:56:43,461 - DEBUG2: Trying to find a VLAN based on the VLAN id '27'
2021-02-12 13:56:43,463 - DEBUG2: Found a exact matching VLAN object: 27 (Chicago, IL)
2021-02-12 13:56:43,463 - DEBUG: IP address 'fe80::250:56ff:fe6a:13ca/64' for vmk4 is a link local address. Skipping.
2021-02-12 13:56:43,463 - DEBUG2: Trying to find a device based on the collected name, cluster, IP and MAC addresses
2021-02-12 13:56:43,472 - DEBUG2: No exact match found. Trying to find device based on MAC addresses
2021-02-12 13:56:43,481 - DEBUG2: No match found. Trying to find device based on serial number
2021-02-12 13:56:43,481 - DEBUG2: Found a matching device object: uslis-p-esx009.example.com (Chicago, IL)
2021-02-12 13:56:43,481 - DEBUG2: Parsing 'device' data structure: uslis-p-esx001.example.com
2021-02-12 13:56:43,482 - DEBUG2: Parsing 'device type' data structure: UCSB-B200-M4
2021-02-12 13:56:43,482 - DEBUG2: Parsing 'manufacturer' data structure: Cisco Systems Inc
2021-02-12 13:56:43,482 - DEBUG: Updated manufacturer object: Cisco Systems Inc
2021-02-12 13:56:43,482 - DEBUG: Updated device type object: UCSB-B200-M4
2021-02-12 13:56:43,482 - DEBUG2: Parsing 'site' data structure: Chicago, IL
2021-02-12 13:56:43,482 - DEBUG: Updated site object: Chicago, IL
2021-02-12 13:56:43,482 - DEBUG2: Parsing 'cluster' data structure: LIS Data Cluster
2021-02-12 13:56:43,482 - DEBUG: Updated cluster object: LIS Data Cluster
2021-02-12 13:56:43,483 - DEBUG2: Parsing 'platform' data structure: VMware ESXi 6.5.0
2021-02-12 13:56:43,483 - DEBUG: Updated platform object: VMware ESXi 6.5.0
2021-02-12 13:56:43,483 - INFO: Device 'uslis-p-esx001.example.com' attribute 'name' changed from 'uslis-p-esx009.example.com' to 'uslis-p-esx001.example.com'
2021-02-12 13:56:43,483 - INFO: Device 'uslis-p-esx001.example.com' attribute 'device_type' changed from 'N20-B6625-1' to 'UCSB-B200-M4'
2021-02-12 13:56:43,483 - DEBUG2: Trying to match current object interfaces in NetBox with discovered interfaces
2021-02-12 13:56:43,491 - DEBUG2: Found '11' NICs in Netbox for 'uslis-p-esx001.example.com'
2021-02-12 13:56:43,491 - DEBUG2: Found 1:1 name match for NIC 'vmnic0'
2021-02-12 13:56:43,491 - DEBUG2: Found 1:1 name match for NIC 'vmnic1'
2021-02-12 13:56:43,491 - DEBUG2: Found 1:1 name match for NIC 'vmnic2'
2021-02-12 13:56:43,492 - DEBUG2: Found 1:1 name match for NIC 'vmnic3'
2021-02-12 13:56:43,492 - DEBUG2: Found 1:1 name match for NIC 'vmnic4'
2021-02-12 13:56:43,492 - DEBUG2: Found 1:1 name match for NIC 'vmnic5'
2021-02-12 13:56:43,492 - DEBUG2: Found 1:1 name match for NIC 'vmk0'
2021-02-12 13:56:43,492 - DEBUG2: Found 1:1 name match for NIC 'vmk1'
2021-02-12 13:56:43,492 - DEBUG2: Found 1:1 name match for NIC 'vmk2'
2021-02-12 13:56:43,492 - DEBUG2: Found 1:1 name match for NIC 'vmk3'
2021-02-12 13:56:43,492 - DEBUG2: Found 1:1 name match for NIC 'vmk4'
2021-02-12 13:56:43,493 - DEBUG2: Parsing 'interface' data structure: vmnic0 (uslis-p-esx001.example.com)
2021-02-12 13:56:43,493 - DEBUG2: Compiling VLAN list
2021-02-12 13:56:43,493 - INFO: Interface 'vmnic0 (uslis-p-esx001.example.com)' attribute 'mac_address' changed from '00:25:B5:01:0A:14' to '00:25:B5:01:0A:17'
2021-02-12 13:56:43,493 - INFO: Interface 'vmnic0 (uslis-p-esx001.example.com)' attribute 'description' changed from '20Gb/s  pNIC (vSwitch0)' to '10Gb/s  pNIC (vSwitch0)'
2021-02-12 13:56:43,493 - INFO: Interface 'vmnic0 (uslis-p-esx001.example.com)' attribute 'type' changed from 'other' to '10gbase-t'
2021-02-12 13:56:43,493 - DEBUG2: Parsing 'interface' data structure: vmnic1 (uslis-p-esx001.example.com)
2021-02-12 13:56:43,494 - DEBUG2: Compiling VLAN list
2021-02-12 13:56:43,494 - INFO: Interface 'vmnic1 (uslis-p-esx001.example.com)' attribute 'mac_address' changed from '00:25:B5:01:0B:14' to '00:25:B5:01:0B:17'
2021-02-12 13:56:43,494 - INFO: Interface 'vmnic1 (uslis-p-esx001.example.com)' attribute 'description' changed from '20Gb/s  pNIC (vSwitch0)' to '10Gb/s  pNIC (vSwitch0)'
2021-02-12 13:56:43,494 - INFO: Interface 'vmnic1 (uslis-p-esx001.example.com)' attribute 'type' changed from 'other' to '10gbase-t'
2021-02-12 13:56:43,494 - DEBUG2: Parsing 'interface' data structure: vmnic2 (uslis-p-esx001.example.com)
2021-02-12 13:56:43,494 - INFO: Interface 'vmnic2 (uslis-p-esx001.example.com)' attribute 'mac_address' changed from '00:25:B5:01:0A:04' to '00:25:B5:01:0A:07'
2021-02-12 13:56:43,494 - INFO: Interface 'vmnic2 (uslis-p-esx001.example.com)' attribute 'description' changed from '20Gb/s  pNIC (DataCluster_dvSwitch)' to '10Gb/s  pNIC (DataCluster_dvSwitch)'
2021-02-12 13:56:43,495 - INFO: Interface 'vmnic2 (uslis-p-esx001.example.com)' attribute 'type' changed from 'other' to '10gbase-t'
2021-02-12 13:56:43,495 - DEBUG2: Parsing 'interface' data structure: vmnic3 (uslis-p-esx001.example.com)
2021-02-12 13:56:43,495 - INFO: Interface 'vmnic3 (uslis-p-esx001.example.com)' attribute 'mac_address' changed from '00:25:B5:01:0B:04' to '00:25:B5:01:0B:07'
2021-02-12 13:56:43,495 - INFO: Interface 'vmnic3 (uslis-p-esx001.example.com)' attribute 'description' changed from '20Gb/s  pNIC (DataCluster_dvSwitch)' to '10Gb/s  pNIC (DataCluster_dvSwitch)'
2021-02-12 13:56:43,495 - INFO: Interface 'vmnic3 (uslis-p-esx001.example.com)' attribute 'type' changed from 'other' to '10gbase-t'
2021-02-12 13:56:43,496 - DEBUG2: Parsing 'interface' data structure: vmnic4 (uslis-p-esx001.example.com)
2021-02-12 13:56:43,496 - INFO: Interface 'vmnic4 (uslis-p-esx001.example.com)' attribute 'mac_address' changed from '00:25:B5:AA:A0:08' to '00:25:B5:AA:A0:00'
2021-02-12 13:56:43,496 - INFO: Interface 'vmnic4 (uslis-p-esx001.example.com)' attribute 'description' changed from '20Gb/s  pNIC (iScsiBootvSwitch)' to '10Gb/s  pNIC (iScsiBootvSwitch)'
2021-02-12 13:56:43,496 - INFO: Interface 'vmnic4 (uslis-p-esx001.example.com)' attribute 'type' changed from 'other' to '10gbase-t'
2021-02-12 13:56:43,496 - DEBUG2: Parsing 'interface' data structure: vmnic5 (uslis-p-esx001.example.com)
2021-02-12 13:56:43,496 - INFO: Interface 'vmnic5 (uslis-p-esx001.example.com)' attribute 'mac_address' changed from '00:25:B5:BB:B0:08' to '00:25:B5:BB:B0:00'
2021-02-12 13:56:43,497 - INFO: Interface 'vmnic5 (uslis-p-esx001.example.com)' attribute 'description' changed from '20Gb/s  pNIC (vSwitch1)' to '10Gb/s  pNIC (vSwitch1)'
2021-02-12 13:56:43,497 - INFO: Interface 'vmnic5 (uslis-p-esx001.example.com)' attribute 'type' changed from 'other' to '10gbase-t'
2021-02-12 13:56:43,497 - DEBUG2: Parsing 'interface' data structure: vmk0 (uslis-p-esx001.example.com)
2021-02-12 13:56:43,497 - INFO: Interface 'vmk0 (uslis-p-esx001.example.com)' attribute 'mac_address' changed from '00:25:B5:01:0A:14' to '00:25:B5:01:0A:17'
2021-02-12 13:56:43,497 - INFO: Interface 'vmk0 (uslis-p-esx001.example.com)' attribute 'description' changed from 'Management VMkernel Port (vSwitch0, vlan ID: 20)' to 'Management VMKernel Port (vSwitch0, vlan ID: 20)'
2021-02-12 13:56:43,497 - DEBUG2: Trying to find prefix for IP: 172.17.5.181/24
2021-02-12 13:56:43,498 - DEBUG2: No matching prefix found for '172.17.5.181/24'
2021-02-12 13:56:43,507 - DEBUG2: Found existing NetBox IP address object: 172.17.5.181/24
2021-02-12 13:56:43,508 - DEBUG2: Parsing 'IP address' data structure: 172.17.5.181/24
2021-02-12 13:56:43,508 - DEBUG2: Parsing 'interface' data structure: vmk1 (uslis-p-esx001.example.com)
2021-02-12 13:56:43,508 - INFO: Interface 'vmk1 (uslis-p-esx001.example.com)' attribute 'mac_address' changed from '00:25:B5:AA:A0:08' to '00:25:B5:AA:A0:00'
2021-02-12 13:56:43,508 - DEBUG2: Trying to find prefix for IP: 10.5.1.11/24
2021-02-12 13:56:43,509 - DEBUG2: No matching prefix found for '10.5.1.11/24'
2021-02-12 13:56:43,516 - DEBUG2: Found existing NetBox IP address object: 10.5.1.11/24
2021-02-12 13:56:43,516 - DEBUG2: Parsing 'IP address' data structure: 10.5.1.11/24
2021-02-12 13:56:43,516 - DEBUG2: Parsing 'interface' data structure: vmk2 (uslis-p-esx001.example.com)
2021-02-12 13:56:43,516 - INFO: Interface 'vmk2 (uslis-p-esx001.example.com)' attribute 'mac_address' changed from '00:50:56:6B:6A:57' to '00:50:56:6B:0E:6E'
2021-02-12 13:56:43,516 - DEBUG2: Trying to find prefix for IP: 10.5.2.11/24
2021-02-12 13:56:43,517 - DEBUG2: No matching prefix found for '10.5.2.11/24'
2021-02-12 13:56:43,541 - DEBUG2: Found existing NetBox IP address object: 10.5.2.11/24
2021-02-12 13:56:43,541 - DEBUG2: Parsing 'IP address' data structure: 10.5.2.11/24
2021-02-12 13:56:43,541 - DEBUG2: Parsing 'interface' data structure: vmk3 (uslis-p-esx001.example.com)
2021-02-12 13:56:43,541 - INFO: Interface 'vmk3 (uslis-p-esx001.example.com)' attribute 'mac_address' changed from '00:50:56:68:61:27' to '00:50:56:6F:5E:3D'
2021-02-12 13:56:43,541 - INFO: Interface 'vmk3 (uslis-p-esx001.example.com)' attribute 'description' changed from 'Fault Tolerance VMkernel Port (vSwitch0, vlan ID: 27)' to 'vMotion VMkernel Port (vSwitch0, vlan ID: 26)'
2021-02-12 13:56:43,542 - INFO: Interface 'vmk3 (uslis-p-esx001.example.com)' attribute 'untagged_vlan' changed from '27 (Chicago, IL)' to '26 (Chicago, IL)'
2021-02-12 13:56:43,542 - DEBUG2: Trying to find prefix for IP: 10.2.16.27/25
2021-02-12 13:56:43,543 - DEBUG2: Found IP '10.2.16.27/25' matches global prefix '10.2.0.0/16'
2021-02-12 13:56:43,543 - WARNING: IP prefix length of '10.2.16.27/25' (vmk3 (uslis-p-esx001.example.com)) does not match network prefix length '10.2.0.0/16'!
2021-02-12 13:56:43,545 - DEBUG2: Found existing NetBox IP address object: 10.2.16.27/25
2021-02-12 13:56:43,545 - DEBUG2: Parsing 'IP address' data structure: 10.2.16.27/25
2021-02-12 13:56:43,546 - DEBUG2: Parsing 'interface' data structure: vmk4 (uslis-p-esx001.example.com)
2021-02-12 13:56:43,546 - INFO: Interface 'vmk4 (uslis-p-esx001.example.com)' attribute 'mac_address' changed from '00:50:56:60:9E:C8' to '00:50:56:6A:13:CA'
2021-02-12 13:56:43,546 - INFO: Interface 'vmk4 (uslis-p-esx001.example.com)' attribute 'description' changed from 'vMotion VMkernel Port (vSwitch0, vlan ID: 26)' to 'Fault Tolerance VMkernel Port (vSwitch0, vlan ID: 27)'
2021-02-12 13:56:43,546 - INFO: Interface 'vmk4 (uslis-p-esx001.example.com)' attribute 'untagged_vlan' changed from '26 (Chicago, IL)' to '27 (Chicago, IL)'
2021-02-12 13:56:43,546 - DEBUG2: Trying to find prefix for IP: 10.2.16.155/25
2021-02-12 13:56:43,547 - DEBUG2: Found IP '10.2.16.155/25' matches global prefix '10.2.0.0/16'
2021-02-12 13:56:43,548 - WARNING: IP prefix length of '10.2.16.155/25' (vmk4 (uslis-p-esx001.example.com)) does not match network prefix length '10.2.0.0/16'!
2021-02-12 13:56:43,550 - DEBUG2: Found existing NetBox IP address object: 10.2.16.155/25
2021-02-12 13:56:43,550 - DEBUG2: Parsing 'IP address' data structure: 10.2.16.155/25
```

**Comment by bb-Ricardo** [comment]
Hi,

the script tries to find a existing device matching different attributes in NetBox. Here it matched the host via serial number.

```
13:56:43,463 - DEBUG2: Trying to find a device based on the collected name, cluster, IP and MAC addresses
2021-02-12 13:56:43,472 - DEBUG2: No exact match found. Trying to find device based on MAC addresses
2021-02-12 13:56:43,481 - DEBUG2: No match found. Trying to find device based on serial number
2021-02-12 13:56:43,481 - DEBUG2: Found a matching device object: uslis-p-esx009.example.com (Chicago, IL)
```

Is it possible that all hosts report the same serial number?

**Comment by justinhippen** [comment]
I dug into this and found that for some reason it is actually reporting the serial number of the blade chassis instead of the server. When I look at the host info in vCenter it reports the correct serial number though. 

I did some digging though the managed object browser and this is what I found.

For the blade servers, it lists the server serial number twice in the HostSystemIdentificationInfo as array value 1 & 3. It also lists the enclosure serial number as array value 2.
![2021-02-13 14_18_12-Managed Object Browser](https://user-images.githubusercontent.com/20194814/107860799-19b43900-6e07-11eb-80be-929c784ebf61.png)

For a "pizza box" server, it lists the server serial number in HostSystemIdentificationInfo array value 2.
![2021-02-13 14_20_21-Managed Object Browser](https://user-images.githubusercontent.com/20194814/107860809-42d4c980-6e07-11eb-91ae-f985b57fce59.png)

I don't know how that matches up with what pyVmomi pulls or if there is even a good way to get around this.

**Comment by bb-Ricardo** [comment]
Very good, thanks for all this digging.

Can you try changing the order of the list here and see if this helps?
https://github.com/bb-Ricardo/netbox-sync/blob/b6c3e990e26922db02b93864dd9d225865e685bd/module/sources/vmware/connection.py#L1476

You might need to delete the device in NetBox first.

Thank you.

**Comment by justinhippen** [comment]
Moving "EnclosureSerialNumberTag" to the back of the list fixed the issue & all hosts were imported.

**Comment by bb-Ricardo** [comment]
Wow, that was easy.

Thank you for testing. Will change it soon in the main branch.

**Comment by lkopyt** [comment]
Hi,
similar but not the same situation here.
during a sync a cluster is filled with only one host device. Semms that devices are replacing each other during scanning and finally only last one is left in inventory.

a cluster with 2 hosts: esxi3-fujitsu and esxi4-fujitsu for instance:


```
2021-02-24 11:44:11,161 - DEBUG2: Trying to find a device based on the collected name, cluster, IP and MAC addresses
2021-02-24 11:44:11,481 - DEBUG2: No exact match found. Trying to find device based on MAC addresses
2021-02-24 11:44:11,702 - DEBUG2: No match found. Trying to find device based on serial number
2021-02-24 11:44:11,703 - DEBUG2: No match found. Trying to find device based on asset tag
2021-02-24 11:44:11,705 - DEBUG2: Found a matching device object: esxi4-fujitsu (PL-WAW-03-LAB01)
2021-02-24 11:44:11,705 - DEBUG2: Parsing 'device' data structure: esxi3-fujitsu
2021-02-24 11:44:11,705 - DEBUG2: Parsing 'device type' data structure: PRIMERGY RX600 S6
2021-02-24 11:44:11,705 - DEBUG2: Parsing 'manufacturer' data structure: FUJITSU
2021-02-24 11:44:11,705 - DEBUG: Updated manufacturer object: FUJITSU
2021-02-24 11:44:11,705 - DEBUG: Updated device type object: PRIMERGY RX600 S6
2021-02-24 11:44:11,705 - DEBUG2: Parsing 'site' data structure: PL-WAW-03-LAB01
2021-02-24 11:44:11,706 - DEBUG: Updated site object: PL-WAW-03-LAB01
2021-02-24 11:44:11,706 - DEBUG2: Parsing 'cluster' data structure: Fujitsu-Cluster
2021-02-24 11:44:11,706 - DEBUG: Updated cluster object: Fujitsu-Cluster
2021-02-24 11:44:11,706 - DEBUG2: Parsing 'platform' data structure: VMware ESXi 6.0.0
2021-02-24 11:44:11,706 - DEBUG: Updated platform object: VMware ESXi 6.0.0
2021-02-24 11:44:11,706 - INFO: Device 'esxi3-fujitsu' attribute 'name' changed from 'esxi4-fujitsu' to 'esxi3-fujitsu'
2021-02-24 11:44:11,706 - INFO: Device 'esxi3-fujitsu' attribute 'serial' changed from 'YLCE007614' to 'YLCE003245'
[...]
2021-02-24 11:44:16,393 - DEBUG2: Trying to find a device based on the collected name, cluster, IP and MAC addresses
2021-02-24 11:44:16,708 - DEBUG2: No exact match found. Trying to find device based on MAC addresses
2021-02-24 11:44:16,956 - DEBUG2: No match found. Trying to find device based on serial number
2021-02-24 11:44:16,957 - DEBUG2: No match found. Trying to find device based on asset tag
2021-02-24 11:44:16,959 - DEBUG2: Found a matching device object: esxi3-fujitsu (PL-WAW-03-LAB01)
2021-02-24 11:44:16,959 - DEBUG2: Parsing 'device' data structure: esxi4-fujitsu
2021-02-24 11:44:16,959 - DEBUG2: Parsing 'device type' data structure: PRIMERGY RX600 S6
2021-02-24 11:44:16,959 - DEBUG2: Parsing 'manufacturer' data structure: FUJITSU
2021-02-24 11:44:16,959 - DEBUG: Updated manufacturer object: FUJITSU
2021-02-24 11:44:16,959 - DEBUG: Updated device type object: PRIMERGY RX600 S6
2021-02-24 11:44:16,959 - DEBUG2: Parsing 'site' data structure: PL-WAW-03-LAB01
2021-02-24 11:44:16,960 - DEBUG: Updated site object: PL-WAW-03-LAB01
2021-02-24 11:44:16,960 - DEBUG2: Parsing 'cluster' data structure: Fujitsu-Cluster
2021-02-24 11:44:16,960 - DEBUG: Updated cluster object: Fujitsu-Cluster
2021-02-24 11:44:16,960 - DEBUG2: Parsing 'platform' data structure: VMware ESXi 6.0.0
2021-02-24 11:44:16,960 - DEBUG: Updated platform object: VMware ESXi 6.0.0
2021-02-24 11:44:16,960 - INFO: Device 'esxi4-fujitsu' attribute 'name' changed from 'esxi3-fujitsu' to 'esxi4-fujitsu'
2021-02-24 11:44:16,960 - INFO: Device 'esxi4-fujitsu' attribute 'serial' changed from 'YLCE003245' to 'YLCE007614'
```

would you please give any hint ?

**Comment by bb-Ricardo** [comment]
Hi,

the important line here is:
```
2021-02-24 11:44:11,705 - DEBUG2: Found a matching device object: esxi4-fujitsu (PL-WAW-03-LAB01)
2021-02-24 11:44:16,959 - DEBUG2: Found a matching device object: esxi3-fujitsu (PL-WAW-03-LAB01)
```
both server have the same asset tag which is totally ok.

Can you try switch off this option and try again?
https://github.com/bb-Ricardo/netbox-sync/blob/208bf73bb4b2bf728e21f0abfc0b8d3924e6bec0/settings-example.ini#L165-L166

**Comment by lkopyt** [comment]
that's right, both hosts had the same factory-default asset tag ("System Asset Tag"), so apparently the scan assumes it found already existing device in inventory and keeps replacing them.
initially I just set different asset tags to each host directly . but somehow sync done on vcenter still assigned old common tag to device (vcenter storing it somewhere in its cache ?).
Finally I disabled the asset_tag option as advised to skip this part, and seems to work fine.

BTW, earlier advice for justinhippen also worked fine for me and my blades in HP BladeCenter.

Thanks!
